src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx

```diff
@@ -612,7 +612,7 @@ const TimeEntryForm = props => {
                   id="dateOfWork"
                   value={formValues.dateOfWork}
                   onChange={handleInputChange}
-                  disabled={from === 'Timer' || !canEditTimeEntry}
+                  disabled={from === 'Timer' || !(canEditTimeEntry || canEditTimelogDate)}
```

Comment:
```suggestion
                  disabled={from === 'Timer' || !canEditTimelogDate}
```
This permission is to replace part of the generic editTimeEntry permission with a more specific permission.